### PR TITLE
Watchlist enhancement

### LINKED ISSUE #80: Watchlist feature enhancement
As a user, I want to be able to see more stock fundamental data apart from the latest prices. (Eg. high low and spread as well as the market) 

cap watchlist to 10 stocks

do not allow adding the same stock again

### PR DESCRIPTION
**Pull request by wwweert123**
fixes #80

### CODE REVIEW

File: src/main/java/seedu/financialplanner/investments/Stock.java [hunk 1]
```diff
@@ -100,4 +101,28 @@ public String getPrice() {
...
     }
+
+    public String getExchange() {
+        return exchange;
+    }
+
+    public void setExchange(String exchange) {
+        this.exchange = exchange;
+    }
+
```
Comment (ryan1604) [inline src/main/java/seedu/financialplanner/investments/Stock.java:112]: I like the clear method names!

File: src/main/java/seedu/financialplanner/investments/WatchList.java [hunk 2]
```diff
@@ -87,16 +86,49 @@ public void fetchFMPStockPrices() throws FinancialPlannerException {
...
         JSONArray ja = (JSONArray) obj;
+        if (ja.size() != stocks.size()) {
+            throw new FinancialPlannerException("Error getting API info!");
+        }
         int i = 0;
         for (Object jo : ja) {
             JSONObject stock = (JSONObject) jo;
-            String price = StringUtils.rightPad(stock.get("price").toString(), 10);
+
+            if (!stocks.get(i).getSymbol().equals(stock.get("symbol"))) {
```
Comment (ryan1604) [inline src/main/java/seedu/financialplanner/investments/WatchList.java]: Maybe add a comment here to explain what this condition means?

File: src/main/java/seedu/financialplanner/investments/WatchList.java [hunk 3]
```diff
@@ -87,16 +86,49 @@ public void fetchFMPStockPrices() throws FinancialPlannerException {
...
+            assert exchange != null;
+            stocks.get(i).setExchange(exchange);
+
+            String dayHigh = stock.get("dayHigh").toString();
+            assert dayHigh != null;
+            stocks.get(i).setDayHigh(dayHigh);
+
+            String dayLow = stock.get("dayLow").toString();
+            assert dayLow != null;
+            stocks.get(i).setDayLow(dayLow);
```
Comment (ryan1604) [inline src/main/java/seedu/financialplanner/investments/WatchList.java]: The fetchFMPStockPrices method seems to be getting a little long, maybe some lines can be extracted into another method.

**Review by ryan1604 — APPROVED** [review]
LGTM

File: src/main/java/seedu/financialplanner/investments/Stock.java [hunk 4]
```diff
@@ -100,4 +101,28 @@ public String getPrice() {
...
+        this.dayHigh = dayHigh;
+    }
+
+    public String getDayLow() {
+        return dayLow;
+    }
+
+    public void setDayLow(String dayLow) {
+        this.dayLow = dayLow;
+    }
```
Comment (NeoMinWei) [inline src/main/java/seedu/financialplanner/investments/Stock.java:127]: good use of getter and setters

File: src/main/java/seedu/financialplanner/investments/WatchList.java [hunk 5]
```diff
@@ -87,16 +86,49 @@ public void fetchFMPStockPrices() throws FinancialPlannerException {
             throw new RuntimeException(e);
         }
         JSONArray ja = (JSONArray) obj;
+        if (ja.size() != stocks.size()) {
+            throw new FinancialPlannerException("Error getting API info!");
```
Comment (NeoMinWei) [inline src/main/java/seedu/financialplanner/investments/WatchList.java:90]: consider adding log here

File: src/main/java/seedu/financialplanner/utils/Ui.java [hunk 6]
```diff
@@ -61,9 +71,12 @@ public void printWatchListHeader() {
...
             String symbol = StringUtils.rightPad(stock.getSymbol(), 10);
-            String price = StringUtils.rightPad(stock.getPrice(), 10);
-            String name = StringUtils.rightPad((String) stock.getStockName(), 10);
-            System.out.println(symbol + price + name);
+            String market = StringUtils.rightPad(stock.getExchange(), 10);
+            String price = YELLOW + StringUtils.rightPad(stock.getPrice(), 10) + RESET;
+            String dayHigh = GREEN + StringUtils.rightPad(stock.getDayHigh(), 15) + RESET;
+            String dayLow = RED + StringUtils.rightPad(stock.getDayLow(), 14) + RESET;
+            String name = StringUtils.rightPad(stock.getStockName(), 10);
+            System.out.println(symbol + market + price + dayHigh + dayLow + name);
```
Comment (NeoMinWei) [inline src/main/java/seedu/financialplanner/utils/Ui.java:79]: good use of creating new string variables and combining them in a single system output

**Review by NeoMinWei — APPROVED** [review]
LGTM

**Pull request merged**